src/compiler/scala/tools/nsc/typechecker/Infer.scala

```diff
@@ -1106,12 +1107,13 @@ trait Infer extends Checkable {
     // this is quite nasty: it destructively changes the info of the syms of e.g., method type params
     // (see #3692, where the type param T's bounds were set to > : T <: T, so that parts looped)
     // the changes are rolled back by restoreTypeBounds, but might be unintentionally observed in the mean time
-    def instantiateTypeVar(tvar: TypeVar) {
-      val tparam                    = tvar.origin.typeSymbol
-      val TypeBounds(lo0, hi0)      = tparam.info.bounds
-      val tb @ TypeBounds(lo1, hi1) = instBounds(tvar)
-      val enclCase                  = context.enclosingCaseDef
-      def enclCase_s                = enclCase.toString.replaceAll("\\n", " ").take(60)
+    private[this] def instantiateTypeVar(tvar: TypeVar): Unit = {
+      val tparam     = tvar.origin.typeSymbol
+      val lo0        = tparam.info.lowerBound
+      val hi0        = tparam.info.upperBound
```

Comment:
My understanding is that this is just a field access. Maybe the compiler is good enough to optimise and factor the access to the `info` field, and from that go to the `.lowerBound` and `upperBound` methods; specially since nothing is changing the `info` field.

If not, my rule-of-thumb understanding is that field access is cheaper than allocating an object.

Reply:
`info` is a lot more than a single field access. It is responsible for ensuring that the symbol is completed, and also returning the current (wrt phase travel) info, which might be buried somewhere in a `TypeHistory`. Even in a fast path, it has to read `phase`, `infos`, `validTo`, `current` (in `TypeHistory`), and possibly more. I know @retronym has done some cool stuff to reduce this overhead recently.

That's not to say it's slower than these allocations! `TypeBounds` are hashconsed, so depending on whether the bounds _are_ the info or need to be synthesized from the info, it may be an allocation and then a uniques set lookup.

The upshot is that I have no idea what's best here, and that's why I was wondering if you'd looked into the tradeoff at all.

Reply:
Ok, I was not aware it could be that complex. In any case, factoring out the `tparam.info` is an easy win.